### PR TITLE
[AVFoundation] Backport new API from master.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Diff used to create the PR: https://gist.github.com/mandel-macaque/79e8e909ca3caeace14d199743d3f926

**Comment by mandel-macaque** [comment]
Failure due to a known issue: https://github.com/xamarin/maccore/issues/638

**Comment by spouliot** [comment]
it's a known issue but it also hide the results

**Comment by spouliot** [comment]
build

File: src/avfoundation.cs [hunk 1]
```diff
@@ -3134,6 +3153,7 @@ interface AVCaptureSynchronizedDataCollection
 	interface IAVCaptureDataOutputSynchronizerDelegate {}
 	
 	[NoWatch, NoTV, iOS (11,0)]
+	[NoMac]
```
Comment (spouliot) [inline src/avfoundation.cs]: that's not correct or it's incomplete as it produce
```
Removed Type MonoMac.AVFoundation.AVCaptureDataOutputSynchronizer

Removed Type MonoMac.AVFoundation.AVCaptureDataOutputSynchronizerDelegate

Removed Type MonoMac.AVFoundation.AVCaptureDataOutputSynchronizerDelegate_Extensions

Removed Type MonoMac.AVFoundation.IAVCaptureDataOutputSynchronizerDelegate

```
in our diffs

Comment (spouliot) [reply src/avfoundation.cs]: part of `src/AVFoundation/AVCompat.cs` also needs to be back ported

Comment (mandel-macaque) [reply src/avfoundation.cs]: I back ported that in commit https://github.com/xamarin/xamarin-macios/pull/3605/commits/9eb9a4125ddae3f0233dbe969cb593844f2b2b85 and we still have the API diff issue:

https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/6541/API_diff_(from_stable)/

Comment (mandel-macaque) [reply src/avfoundation.cs]: Ignore me, fixed in commit https://github.com/xamarin/xamarin-macios/pull/3605/commits/12de09ea9213535e9afaea354038be4233289f4e

Clearly not my week :/

**Comment by mandel-macaque** [comment]
Keep getting the xtro failure, we will retry.

**Comment by mandel-macaque** [comment]
build

**Comment by spouliot** [comment]
known issue (xtro) -> https://github.com/xamarin/maccore/issues/638

**Pull request merged**